internal/devbox/devbox.go

```diff
@@ -819,23 +804,34 @@ func (d *Devbox) ensureStateIsUpToDateAndComputeEnv(
 ) (map[string]string, error) {
 	defer debug.FunctionTimer().End()
 
-	// When ensureStateIsUpToDate is called with ensure=true, it always
-	// returns early if the lockfile is up to date. So we don't need to check here
-	if err := d.ensureStateIsUpToDate(ctx, ensure); isConnectionError(err) {
-		if !fileutil.Exists(d.nixPrintDevEnvCachePath()) {
-			ux.Ferrorf(
+	if envOpts.RecomputeEnv.Disabled {
```

Comment:
Changes in `ensureStateIsUpToDateAndComputeEnv` look OK, but please double triple check. This is very core functioning stuff.